manifests/server/service.pp

```diff
@@ -10,10 +10,16 @@
     }
   }
 
+  if $mysql::server::override_options['mysqld'] and $mysql::server::override_options['mysqld']['user'] {
+    $mysqluser = $mysql::server::override_options['mysqld']['user']
+  } else {
+    $mysqluser = $options['mysqld']['user']
+  }
+
   file { $options['mysqld']['log-error']:
     ensure => present,
-    owner  => 'mysql',
-    group  => 'mysql',
+    owner  => $mysqluser,
+    group  => $::mysql::server::root_group,
```

Comment:
@buzzdeee @igalic The group change breaks idempotency on non-debian systems [(example)]( https://www.google.com/url?q=https%3A%2F%2Fjenkins-modules.puppetlabs.com%2Fview%2F1.%2520failing%2520jobs%2Fjob%2Fmodules_puppetlabs-mysql_intn-sys_ext-master-enterprise-stable%2F36%2FPLATFORM_CONFIG%3Dcentos-5-64mda%2CSLAVE_LABEL%3Dbeaker%2Fconsole&sa=D&sntz=1&usg=AFQjCNGvo20G6wjvdWodNjjMEqFtS3Asqg). I'm assuming this change was needed for OpenBSD support, but this should be refactored to leave the group `'mysql'` for other OSes.